arelle/plugin/validate/ROS/rules/ros.py

```diff
@@ -368,13 +368,13 @@ def rule_ros19(
         for unit, count in unit_counts.items():
             if count > principal_currency_value_count:
                 modelObjects = principal_currency_facts
-                message = _("'PrincipalCurrencyUsedInBusinessReport' has a value of %(principal_currency_value)s, "
+                message = _("'PrincipalCurrencyUsedInBusinessReport' has a value of %(principal_currency_values)s, "
                             "which must match the functional(majority) unit of the financial statement.")
                 continue
     if message:
         yield Validation.warning(
             "ROS.19",
             message,
             modelObject=modelObjects,
-            principal_currency_value=principal_currency_value
+            principal_currency_values=principal_currency_values if principal_currency_values else None
```

Comment:
Is the intent to turn empty set into `None`?  If yes, out of curiosity, why?  If no, why not simply `principal_currency_values=principal_currency_values`?